### ISSUE TITLE
Cleanup policy of an event hub fall back to delete from compact after alternating its partition count

### ISSUE THREAD

**Issue by wanwenli** [Azure/azure-event-hubs-for-kafka#167]
Description
===========
All discussions below are in the context of the _Kafka_ interface of Azure event hubs.

An event hub can be created via Kafka shell with `compact` as its cleanup policy. However, later when a user changes its partition count on Azure portal, the cleanup policy of the event hub falls back to `delete`.

How to reproduce
================
Download Kafka executables from the official website.

Create an event hub with `compact` cleanup policy
```
bin/kafka-topics.sh --bootstrap-server <namespace>.servicebus.windows.net:9093 --topic kafka-connect-offsets --command-config config/client_common.properties --create --partitions 3 --replication-factor 1 --config cleanup.policy=compact
```

Change the partition count of the event hub on [Azure portal](https://portal.azure.com/).

Verify the cleanup policy of the event hub.
```
bin/kafka-topics.sh --bootstrap-server <namespace>.servicebus.windows.net:9093 --topic kafka-connect-offsets --command-config config/client_common.properties --describe
```

It's found that the cleanup policy has fallen back to `delete`.

Has it worked previously?
=========================
Never

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed or where adequate information has not been provided.

Please provide the relevant information for the following items:
 - [ ] SDK (include version info): `<REPLACE with e.g., kafka-python/Java SDK/confluent-kafka-dotnet with version info>`
 - [ ] Sample you're having trouble with: `<REPLACE with e.g., Java quickstart>`
 - [x] If using Apache Kafka Java clients or a framework that uses Apache Kafka Java clients, version: `2.4.1`
 - [ ] Kafka client configuration: `<REPLACE with e.g., auto.reset.offset=earliest, ..>` (*do not include your connection string or SAS Key*)
 - [ ] Namespace and EventHub/topic name
 - [ ] Consumer or producer failure `<REPLACE with e.g., Consumer failure>`
 - [ ] Timestamps *in UTC* `<REPLACE with e.g., Nov 7 2018 - 17:15:01 UTC>`
 - [ ] group.id or client.id `<REPLACE with e.g., group.id=cg-name>`
 - [ ] Logs provided (with debug-level logging enabled if possible, e.g. log4j.rootLogger=DEBUG) or exception call stack
 - [ ] Standalone repro `<REPLACE with e.g., Willing/able to send scenario to repro issue>`
 - [ ] Operating system: `<REPLACE with e.g., Ubuntu 16.04.5 (x64) LTS>`
 - [ ] Critical issue

If this is a question on basic functionality, please verify the following:
 - [x] Port 9093 should not be blocked by firewall ("broker cannot be found" errors)
 - [x] Pinging FQDN should return cluster DNS resolution (e.g. `$ ping namespace.servicebus.windows.net` returns ~ `ns-eh2-prod-am3-516.cloudapp.net [13.69.64.0]`)
 - [x] Namespace should be either Standard or Dedicated tier, not Basic (TopicAuthorization errors)

**Comment by HuaqiongWu** [comment]
I encountered the same thing @wanwenli  Did you solve it?

**Comment by wanwenli** [comment]
> I encountered the same thing @wanwenli Did you solve it?

nope, I haven't resolved it yet. probably it's a bug of Azure event hubs.

**Comment by ychen4-li** [comment]
Does log compaction work if we don't alternate partition count? I read somewhere that log compaction is not supported by EH, however Kafka Connect internal topics require log compaction to persistent dynamic config...
